src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.spec.ts

```diff
@@ -51,6 +63,35 @@ describe('DotEditPageWorkflowsActionsComponent', () => {
     it('should set action split buttons params', () => {
         fixture.detectChanges();
         const actionsButton: SplitButton = de.query(By.css('.edit-page-toolbar__actions')).componentInstance;
-        expect(actionsButton.model).toEqual([{ label: 'Assign Workflow' }, { label: 'Save' }, { label: 'Save / Publish' }]);
+
+        expect(actionsButton.model[0].label).toEqual('Assign Workflow');
+        expect(actionsButton.model[1].label).toEqual('Save');
+        expect(actionsButton.model[2].label).toEqual('Save / Publish');
     });
+
+    it('should trigger the methods in the action buttons', () => {
+        const dotWorkflowService: DotWorkflowService = de.injector.get(DotWorkflowService);
+        let workflowsActions: DotWorkflowAction[];
+        dotWorkflowService.getContentWorkflowActions(component.inode).subscribe((event) => (workflowsActions = event));
+        spyOn(dotWorkflowService, 'fireWorkflowAction');
+
+        fixture.detectChanges();
+
+        const splitButtons = de.query(By.all()).nativeElement.querySelectorAll('.ui-menuitem-link');
+        const firstButton = splitButtons[0];
+        const secondButton = splitButtons[1];
+        const thirdButton = splitButtons[2];
+
+        firstButton.click();
+        expect(dotWorkflowService.fireWorkflowAction).toHaveBeenCalledWith(component.inode, workflowsActions[0].id);
+
+        secondButton.click();
+        expect(dotWorkflowService.fireWorkflowAction).toHaveBeenCalledWith(component.inode, workflowsActions[1].id);
+
+        thirdButton.click();
+        expect(dotWorkflowService.fireWorkflowAction).toHaveBeenCalledWith(component.inode, workflowsActions[2].id);
+
+        expect(dotWorkflowService.fireWorkflowAction).toHaveBeenCalledTimes(3);
```

Comment:
We have to check that the 'dotWorkflowService.getContentWorkflowActions' method have to be call a second times after the fireWorkflowAction is returned